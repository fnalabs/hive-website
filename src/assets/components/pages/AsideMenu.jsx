import React from 'react'
import { NavLink } from 'react-router-dom'

import { AnalyticsLink } from 'common'

const AsideMenu = () => (
  <aside className='menu'>
    <p className='menu-label has-text-weight-bold'>v2.1.3</p>
    <ul className='menu-list'>
      <li><NavLink to='/overview' activeClassName='is-active'>Overview</NavLink>
        <ul>
          <li><NavLink to='/model' activeClassName='is-active'>Data Model</NavLink></li>
          <li><NavLink to='/domain' activeClassName='is-active'>Domain Logic</NavLink></li>
          <li><NavLink to='/infrastructure' activeClassName='is-active'>Infrastructure</NavLink></li>
          <li><NavLink to='/telemetry' activeClassName='is-active'>Telemetry</NavLink></li>
        </ul>
      </li>

      <li><NavLink to='/start' activeClassName='is-active'>Get Started</NavLink>
        <ul>
          <li><NavLink to='/setup' activeClassName='is-active'>Setup</NavLink></li>
          <li className='menu-label has-text-weight-bold'>Examples:</li>
          <li><NavLink to='/basic' activeClassName='is-active'>Basic</NavLink></li>
          <li><NavLink to='/rest' activeClassName='is-active'>Rest</NavLink></li>
          <li><NavLink to='/cqrs-es' activeClassName='is-active'>CQRS/ES</NavLink></li>
        </ul>
      </li>

      <li><NavLink to='/documentation' activeClassName='is-active'>Documentation</NavLink>
        <ul>
          <li><NavLink to='/environments' activeClassName='is-active'>Environments</NavLink></li>
          <li><AnalyticsLink to='https://fnalabs.github.io/hive-io/' target='_blank' rel='noopener noreferrer'>API</AnalyticsLink></li>
        </ul>
      </li>
    </ul>
  </aside>
)

export default AsideMenu
