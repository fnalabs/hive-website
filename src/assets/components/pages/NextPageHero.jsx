import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

import { AnalyticsLink } from 'common'
import { Hero, HeroBody } from 'common/hero'
import { ArrowRight } from 'icons'

const toRegex = /^https:\/\//

export default class NextPageHero extends Component {
  static propTypes = {
    to: PropTypes.string.isRequired
  }

  render () {
    const to = this.props.to

    return (
      <Hero size='small' className='is-hive'>
        <HeroBody centered>
          {toRegex.test(to)
            ? <AnalyticsLink to={to}><ArrowRight height='48' /></AnalyticsLink>
            : <Link to={to}><ArrowRight height='48' /></Link>}
        </HeroBody>
      </Hero>
    )
  }
}
