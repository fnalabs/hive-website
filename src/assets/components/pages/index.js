// site's main pages
export { default as Home } from './home/Home.jsx'
export { default as Overview } from './overview/Overview.jsx'
export { default as Start } from './start/Start.jsx'

// policy pages
export { Cookie } from './policies/Cookie.jsx'
export { Privacy } from './policies/Privacy.jsx'
