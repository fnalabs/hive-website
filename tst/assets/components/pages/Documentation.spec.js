/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { MemoryRouter } from 'react-router'
import ReactGA from 'react-ga'

import { Documentation } from 'pages'

jest.mock('react-ga')

describe('<Documentation />', () => {
  it('should render Documentation page static content', () => {
    const tree = renderer.create(<MemoryRouter><Documentation location={{ pathname: '/documentation' }} /></MemoryRouter>).toJSON()

    expect(tree).toMatchSnapshot()
    expect(ReactGA.pageview).not.toBeCalled()
  })

  it('should render Documentation page static content and report page view', () => {
    shallow(<Documentation location={{ pathname: '/documentation' }} />, {
      context: { isConsent: true }
    })

    expect(ReactGA.pageview).toBeCalledTimes(1)
  })
})
