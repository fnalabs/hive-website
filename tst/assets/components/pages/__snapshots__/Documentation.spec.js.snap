// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Documentation /> should render Documentation page static content 1`] = `
Array [
  <article
    className="section is-medium"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <header
          className="column is-narrow is-hidden-touch"
        >
          <aside
            className="menu"
          >
            <p
              className="menu-label has-text-weight-bold"
            >
              v2.1.2
            </p>
            <ul
              className="menu-list"
            >
              <li>
                <a
                  aria-current={null}
                  href="/overview"
                  onClick={[Function]}
                >
                  Overview
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/model"
                      onClick={[Function]}
                    >
                      Data Model
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/domain"
                      onClick={[Function]}
                    >
                      Domain Logic
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/infrastructure"
                      onClick={[Function]}
                    >
                      Infrastructure
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/telemetry"
                      onClick={[Function]}
                    >
                      Telemetry
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/start"
                  onClick={[Function]}
                >
                  Get Started
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/setup"
                      onClick={[Function]}
                    >
                      Setup
                    </a>
                  </li>
                  <li
                    className="menu-label has-text-weight-bold"
                  >
                    Examples:
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/basic"
                      onClick={[Function]}
                    >
                      Basic
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/rest"
                      onClick={[Function]}
                    >
                      Rest
                    </a>
                  </li>
                  <li>
                    <a
                      aria-current={null}
                      href="/cqrs-es"
                      onClick={[Function]}
                    >
                      CQRS/ES
                    </a>
                  </li>
                </ul>
              </li>
              <li>
                <a
                  aria-current={null}
                  href="/documentation"
                  onClick={[Function]}
                >
                  Documentation
                </a>
                <ul>
                  <li>
                    <a
                      aria-current={null}
                      href="/environments"
                      onClick={[Function]}
                    >
                      Environments
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-io/"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      API
                    </a>
                  </li>
                </ul>
              </li>
            </ul>
          </aside>
        </header>
        <section
          className="column content"
          role="document"
        >
          <h1>
            Documentation
          </h1>
          <p>
            Below you will find a sitemap for API documentation of the Hive
            <sup>
              io
            </sup>
             library. There are links to each of the Classes and links to their methods. Also, included on the next page is all of the environment variables for each microservice image.
          </p>
          <h2
            id="api"
          >
            API
          </h2>
          <p>
            <em>
              <strong>
                NOTE:
              </strong>
            </em>
             These pages are currently available online only.
          </p>
          <nav>
            <ul>
              <li>
                <a
                  href="https://fnalabs.github.io/hive-io/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Home
                </a>
                <ul>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-io/Actor.html"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Actor
                    </a>
                    <ul>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Actor.html#assign"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          assign
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Actor.html#perform"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          perform
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Actor.html#replay"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          replay
                        </a>
                      </li>
                    </ul>
                  </li>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-io/MessageActor.html"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      MessageActor
                    </a>
                    <ul>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/MessageActor.html#assign"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          assign
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/MessageActor.html#perform"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          perform
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Actor.html#replay"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          replay
                        </a>
                      </li>
                    </ul>
                  </li>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-io/Model.html"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Model
                    </a>
                    <ul>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Model.html#.errors"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          errors
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Model.html#.schema"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          schema
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Model.html#.validate"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          validate
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Model.html#.version"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          version
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Model.html#toJSON"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          toJSON
                        </a>
                      </li>
                    </ul>
                  </li>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-io/Schema.html"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Schema
                    </a>
                    <ul>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Schema.html#assign"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          assign
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Schema.html#validate"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          validate
                        </a>
                      </li>
                    </ul>
                  </li>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-io/System.html"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      System
                    </a>
                    <ul>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/System.html#emit"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          emit
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/System.html#on"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          on
                        </a>
                      </li>
                    </ul>
                  </li>
                  <li>
                    <a
                      href="https://fnalabs.github.io/hive-io/Bus.html"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Bus
                    </a>
                     (alias of System)
                    <ul>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Bus.html#emit"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          emit
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://fnalabs.github.io/hive-io/Bus.html#on"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          on
                        </a>
                      </li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </nav>
        </section>
      </div>
    </div>
  </article>,
  <section
    className="hero is-small is-hive"
  >
    <div
      className="hero-body"
    >
      <nav
        className="level is-mobile"
      >
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="previous page"
            href="/cqrs-es"
            onClick={[Function]}
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M223.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L319.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L393.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34zm-192 34l136 136c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9L127.9 256l96.4-96.4c9.4-9.4 9.4-24.6 0-33.9L201.7 103c-9.4-9.4-24.6-9.4-33.9 0l-136 136c-9.5 9.4-9.5 24.6-.1 34z"
                />
              </svg>
            </span>
          </a>
        </div>
        <div
          className="level-item has-text-centered"
        >
          <a
            aria-label="next page"
            href="/environments"
            onClick={[Function]}
          >
            <span
              className="icon is-large"
            >
              <svg
                height="48"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34zm192-34l-136-136c-9.4-9.4-24.6-9.4-33.9 0l-22.6 22.6c-9.4 9.4-9.4 24.6 0 33.9l96.4 96.4-96.4 96.4c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l136-136c9.4-9.2 9.4-24.4 0-33.8z"
                />
              </svg>
            </span>
          </a>
        </div>
      </nav>
    </div>
  </section>,
]
`;
